design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,147 @@
+- Reusable. Device discovery can be used for other storage backends.
+
+
+### Detailed Device Discovery Process
+
+Each `Device Discovery` Pod reads device filters from `Device Filter Configmap`, retrieves named device filters, walks through device trees and uses device filters to identify the devices, and stores the filtering results in `Raw Device Configmap`. 
+
+#### Device Filter Configmap
+
+This Configmap defines the node regex and device filters on each node. 
+
+A sample of filter Configmap is as the following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: rook-system
+  name: device-filters
+data:
+  config: |
+    - nodes: "foo*" # node name regex
+      filters: 
+      - name: "search sd*"
+        filter: by-device-path
+        parameters: 
+          - "/dev/sd*"
+    - nodes: "bar*" # node name regex
+      filters: 
+      - name: "search big disks"
+        filter: by-device-capacity
+        parameters:
+           - "gt" # greater than
+           - "100GiB" # 100GiB 
+      - name: "search vendor X"
+        filter: by-device-vendor
+        parameters:
+          - "X" # vendor name
+```
+
+In this configuration, device filters are applied to node names with match regex. Each named device filters, `by-device-path`, `by-device-capacity`, and `by-device-vendor`, are implemented by the Device Discovery Pod to conduct certain search criteria. For instance, filter `by-device-capacity` is implemented to probe each block device's and filters based on capacity given in the parameter, which in this example is `100GiB`. 
+
+Each of the named filter yields a formatted output. In the `by-device-capacity` filter, the output is a json string like the following. Note `device-path` is mandatory for all filter output.
+
+```json
+{
+ "output": [ 
+   {"device-path": "/dev/sdx", "capacity": 214748364800},
+   {"device-path": "/dev/sdy", "capacity": 214748364800}
+ ]
+}
+```
+
+#### Raw Device Configmap
+
+After devices are discovered, they are stored in `Raw Device Configmap`. A sample is as the following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: rook-system
+  name: raw-devices
+data:
+  devices: |
+    - nodes: "foo-node1"
+      - name: "search big disks"
+        filter: by-device-capacity
+        result: '{ "output": {"device-path": "/dev/sdx", "capacity": 214748364800}, {"device-path": "/dev/sdy", "capacity": 214748364800} }'
+```
+
+#### Adding and Removing Device Filters
+
+Device filters are updated whenever a cluster is deleted or new clusters are created. Operator updates `Device Filter Configmap` and `Raw Device Configmap` using filter names and node regex.
+ 
+### Discussions
+
+It is expected Device Discovery will be merged into Rook Operator once local PVs are supported in Rook Cluster CRD. Rook Operator can infer the device topoloyg from local PV Configmaps. However, as long as raw devices or directories are still in use, a dedicated Device Discovery Pod is still needed.
```

Comment:
`topoloyg` is misspelled, as @jbw976 pointed out in a previous comment

Reply:
done